webpack/assets/javascripts/react_app/components/charts/Chart.jsx

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import Component from '../common/BindingComponent';
+
+import CenteredLoader from '../common/CenteredLoader';
+export default class extends Component {
+    constructor(props) {
+        super(props);
+    }
+
+    render() {
+        let chart;
+
+        if (this.props.isLoaded) {
+            this.props.drawChart();
+            chart = (<div className={this.props.cssClass} id={this.props.id + 'Chart'}></div>);
+        } else {
```

Comment:
note that isLoaded = true but hasData is false, so the spinner disappear very quickly before data is received.